src/storage/mvcc/reader/scanner/mod.rs

```diff
@@ -154,3 +143,47 @@ impl<S: Snapshot> Scanner<S> {
         }
     }
 }
+
+pub struct ScannerConfig<S: Snapshot> {
+    pub snapshot: S,
+    pub fill_cache: bool,
+    pub omit_value: bool,
+    pub isolation_level: IsolationLevel,
+
+    /// `lower_bound` and `upper_bound` is used to create `default_cursor`. `upper_bound`
+    /// is used in initial seek as well. They will be consumed after `default_cursor` is being
```

Comment:
... `upper_bound` is used in initial forward seek (or `lower_bound` in initial backward seek) as well ...